File: THCS/2022_2023/TienGiang_2223/Bai2/STRING.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <algorithm>
#include <cctype>

using namespace std;

int main() {
    ifstream input("STRING.INP");
    ofstream output("STRING.OUT");

    string s;
    getline(input, s); // Đọc xâu từ tệp

    string letters; // Chứa các ký tự chữ
    string digits; // Chứa các ký tự số

    // Phân tách ký tự thành chữ cái và số
    for (char ch : s) {
        if (isalpha(ch)) {
            letters += ch; // Thêm chữ cái vào letters
        } else if (isdigit(ch)) {
            digits += ch; // Thêm chữ số vào digits
        }
    }

    // Đảo ngược chuỗi letters
    reverse(letters.begin(), letters.end());

    // Sắp xếp chuỗi digits
    sort(digits.begin(), digits.end());

    // Ghi kết quả vào tệp
    output << letters << endl;
    output << digits << endl;

    input.close();
    output.close();
    return 0;
}

File: THCS/2022_2023/TienGiang_2223/Bai2/STRING.py
# Đọc dữ liệu từ tệp STRING.INP
with open('STRING.INP', 'r') as input_file:
    s = input_file.readline().strip()  # Đọc xâu và loại bỏ ký tự thừa

# Khởi tạo hai danh sách để lưu ký tự chữ và số
letters = []
digits = []

# Phân tách ký tự thành chữ cái và số
for ch in s:
    if ch.isalpha():
        letters.append(ch)  # Thêm chữ cái vào danh sách letters
    elif ch.isdigit():
        digits.append(ch)  # Thêm chữ số vào danh sách digits

# Đảo ngược danh sách letters và sắp xếp danh sách digits
letters.reverse()
digits.sort()

# Ghi kết quả vào tệp STRING.OUT
with open('STRING.OUT', 'w') as output_file:
    output_file.write(''.join(letters) + '\n')  # Ghi các chữ cái
    output_file.write(''.join(digits) + '\n')    # Ghi các chữ số
